### PR TITLE
Add dartdoc for image classes

### LINKED ISSUE #361: Multiple "Image" classes is confusing
There's one Image class in flutter/lib/src/widgets/basic.dart
There's another Image class in dart:ui.

The first wraps the second.

Since I just spent an hour sorting this problem, it seems that overloading this class name will cause others confusion too.

### PR DESCRIPTION
**Pull request by abarth**
- Adds dartdoc for all the `of` functions.
  - Renames Image to RawImage. This widget is rarely used and shouldn't take up
    such a nice global name.

Fixes #361

### CODE REVIEW

File: packages/flutter/lib/src/widgets/basic.dart [hunk 1]
```diff
@@ -1365,13 +1370,40 @@ class Image extends LeafRenderObjectWidget {
...
+  /// aspect ratio.
   final double height;
+
+  /// If non-null, apply this color filter to the image before painint.
   final ColorFilter colorFilter;
+
+  /// How to inscribe the image into the place allocated during layout.
   final ImageFit fit;
+
+  /// How to align the image within its bounds.
```
Comment (Hixie) [inline packages/flutter/lib/src/widgets/basic.dart]: This should have more detail since the definition of FractionalOffset probably can't help.

Comment (Hixie) [inline packages/flutter/lib/src/widgets/basic.dart]: (same comment for other classes with this field)

File: packages/flutter/lib/src/widgets/routes.dart [hunk 2]
```diff
@@ -384,6 +384,7 @@ abstract class ModalRoute<T> extends TransitionRoute<T> with LocalHistoryRoute<T
 
   final RouteSettings settings;
 
+  /// The closest instance of this class that encloses the given context.
   static ModalRoute of(BuildContext context) {
```
Comment (Hixie) [inline packages/flutter/lib/src/widgets/routes.dart]: technically the routes don't enclose anything, they're off in a different world. It's the route for the closest ModalScope, but ModalScopes are an implementation detail. Dunno how to explain that here though...

File: packages/flutter_sprites/lib/src/texture.dart [hunk 3]
```diff
@@ -2,10 +2,10 @@ part of flutter_sprites;
 
 /// A texture represents a rectangular area of an image and is typically used to draw a sprite to the screen.
 ///
-/// Normally you get a reference to a texture from a [SpriteSheet], but you can also create one from an [Image].
+/// Normally you get a reference to a texture from a [SpriteSheet], but you can also create one from an [RawImage].
 class Texture {
 
-  /// Creates a new texture from an [Image] object.
+  /// Creates a new texture from an [RawImage] object.
```
Comment (Hixie) [inline packages/flutter_sprites/lib/src/texture.dart]: nonono, this is talking about the _other_ Image.

**Comment by Hixie** [comment]
We sure have an abundance of image-related widgets.

**Comment by Hixie** [comment]
LGTM

File: packages/flutter/lib/src/widgets/routes.dart (hunk 2 above)
Comment (abarth) [inline packages/flutter/lib/src/widgets/routes.dart]: Yeah, nested navigators make it especially tricky to describe.  Maybe we should read "enclosing" to not refer strictly to the parent relation among widgets?  Maybe there's a better word to use that indicates that his one is different.

File: packages/flutter_sprites/lib/src/texture.dart (hunk 3 above)
Comment (abarth) [inline packages/flutter_sprites/lib/src/texture.dart]: Good point.  :)

**Comment by abarth** [comment]
> We sure have an abundance of image-related widgets.

Yes.  Now that we have ImageProvider, I bet we could merge RawImageResource and AsyncImage.  Then we could layer AssetImage and NetworkImage on top of AsyncImage.

File: packages/flutter/lib/src/widgets/routes.dart (hunk 2 above)
Comment (abarth) [inline packages/flutter/lib/src/widgets/routes.dart]: I went with the vague "associated" term.

**Pull request merged**